exchanges/request/request.go

```diff
@@ -150,7 +150,22 @@ func (r *Requester) doRequest(ctx context.Context, endpoint EndpointLimit, newRe
 			}
 		}
 
+		reporter := GlobalReporter
+		if reporter == nil {
+			// no global reporter defined, fallback to the request's
+			reporter = r.reporter
+			if reporter == nil {
+				// no custom reporter as well, fallback to the default
+				// that will look at the Verbose flag to decide wether
+				// to log or not
+				reporter = DefaultReporter(p.Verbose)
+			}
+		}
```

Comment:
This seems like it would be better to do this at a setup/requester setup. When no reporters are defined, we are doing multiple checks and creating structs on every request.

Reply:
i ended up dropping the DefaultReporter to simplify this logic, if an exchange defines a reporter then that takes precedence over a globally defined one as well.